backend/de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/warehouse/api/IWarehouseDAO.java

```diff
@@ -205,4 +205,16 @@ private WarehouseQuery(
 	WarehouseId retrieveWarehouseIdBy(WarehouseQuery query);
 
 	WarehouseAndLocatorValue retrieveWarehouseAndLocatorValueByLocatorRepoId(int locatorRepoId);
+
+	@NonNull
+	Optional<WarehouseId> getOptionalIdByValue(@NonNull String value);
+
+	@NonNull
+	Optional<Warehouse> getOptionalById(@NonNull WarehouseId id);
+
+	void updateWarehouse(@NonNull Warehouse request);
```

Comment:
imho this method can be simply called `save` 
and `request` can be called `warehouse`